frame/nomination-pools/src/lib.rs

```diff
@@ -582,25 +583,259 @@ pub struct PoolRoles<AccountId> {
+/// If a `throttle` is set before a commission `current` is set, the initial commission `current`
+/// value will not be subject to throttling. Subsequent commission updates will be.
+#[derive(
+	Encode, Decode, DefaultNoBound, MaxEncodedLen, TypeInfo, DebugNoBound, PartialEq, Copy, Clone,
+)]
+#[codec(mel_bound(T: Config))]
+#[scale_info(skip_type_params(T))]
+pub struct Commission<T: Config> {
+	/// Optional commission rate of the pool along with the account commission is paid to.
+	pub current: Option<(Perbill, T::AccountId)>,
+	/// Optional maximum commission that can be set by the pool `root`. Once set, this value
+	/// can only be updated to a decreased value.
+	pub max: Option<Perbill>,
+	/// Optional configuration around how often commission can be updated, and when the last
+	/// commission update took place.
+	pub throttle: Option<CommissionThrottle<T>>,
+}
+
+impl<T: Config> Commission<T> {
+	/// Get the current commission percentage of this pool.
+	///
+	/// Returns zero if commission has not yet been configured.
+	fn commission_or_zero(&self) -> Perbill {
+		self.current.as_ref().map(|(x, _)| *x).unwrap_or(Perbill::zero())
+	}
+
+	/// Returns true if a commission percentage updating to `to` would exhaust the throttle limit.
+	///
+	/// A commission update will be throttled (disallowed) if:
+	/// 1. not enough blocks have passed since the previous commission update took place, and
+	/// 2. the new commission is larger than the maximum allowed increase.
+	///
+	/// Throttling is not applied to commission updates if `current` is still `None`.
+	fn throttling(&self, to: &Perbill) -> bool {
+		if let Some(t) = self.throttle.as_ref() {
+			// factor previously updated block into whether user is throttled.
+			if t.previous_set_at.map_or(
+				// if no `previous_set_at` is set, throttled if the attempted increase
+				// in commission is greater than `max_increase`.
+				(*to).saturating_sub(self.commission_or_zero()) > t.change_rate.max_increase,
+				|p| {
+					// `min_delay` blocks must have been surpassed since last update.
+					if <frame_system::Pallet<T>>::block_number().saturating_sub(p) <
+						t.change_rate.min_delay
+					{
+						return true
+					}
+					// ensure the `max_increase` durations surpassed since the previous
+					// commission update allow the attempted commission increase.
+					//
+					// the attempted increase in commission relative to the current commission.
+					let attempted_increase = (*to).saturating_sub(self.commission_or_zero());
+
+					// the total durations passed since the last commission update.
+					let intervals_passed = self.intervals_since_block(&p, &t.change_rate.min_delay);
+
+					// the maximum allowed increase, where the current `max_increase`
+					// Perbill is converted into a u32 by multiplying itself with 100_u32
+					// and multiplied by durations passed, before being converted back
+					// into a Perbill.
+					let max_allowed_increase = Perbill::from_percent(
```

Comment:
This will allow pool operator to still abruptly increase pool commission to a really high value after waiting a long time. Not sure if thats desired?

Reply:
We could also just simplify this and `max_increase` is the `max_allowed_increase` irrespective of number of intervals passed.

Reply:
> This will allow pool operator to still abruptly increase pool commission to a really high value after waiting a long time. Not sure if thats desired?

This is the concern I had before, but the solution is to have a max commission set so it becomes a bit more predictable.

Reply:
Yeah I was of the opinion that treating is an interval is actually more intuitive. We can revise later.